tests/cpp_unit_tests/test_sparse_mapping.cpp

```diff
@@ -17,4 +18,58 @@ TEST_CASE("Test sparse mapping") {
     CHECK(mapping.reorder == mapping_2.reorder);
 }
 
+TEST_CASE("Test dense mapping - n log n sort") {
+    IdxVector const idx_B_in_A{3, 5, 2, 1, 1, 2};
+    DenseMapping mapping{{1, 1, 2, 2, 3, 5}, {3, 4, 2, 5, 0, 1}};
+    DenseMapping mapping_2 = build_dense_mapping(idx_B_in_A, 7);
+
+    CHECK(mapping.indvector == mapping_2.indvector);
+    CHECK(mapping.reorder == mapping_2.reorder);
+    CHECK(*(mapping_2.indvector.begin()) == *std::min_element(mapping_2.indvector.begin(), mapping_2.indvector.end()));
+    CHECK(*(mapping_2.indvector.end() - 1) ==
+          *std::max_element(mapping_2.indvector.begin(), mapping_2.indvector.end()));
+}
+
+// TEST_CASE("Test dense mapping - comparison sort") {
+
+//     IdxVector idx_B_in_A(1000000);
+//     std::mt19937 eng(16);
+//     std::uniform_int_distribution<> distr(0, 1000000);
+
+//     for (auto& val : idx_B_in_A) {
+//         val = distr(eng);
+//     }
+
+//     IdxVector sorted_idx_B_in_A = idx_B_in_A;
+//     std::ranges::sort(sorted_idx_B_in_A);
+
+//     DenseMapping mapping_2 = build_dense_mapping(idx_B_in_A, 1000001);
+
+//     CHECK(mapping_2.indvector == sorted_idx_B_in_A);
+//     CHECK(*(mapping_2.indvector.begin()) == *std::min_element(mapping_2.indvector.begin(),
+//     mapping_2.indvector.end())); CHECK(*(mapping_2.indvector.end() - 1) ==
+//           *std::max_element(mapping_2.indvector.begin(), mapping_2.indvector.end()));
+// }
+
```

Comment:
for unit tests, please don't randomly generate, OR fix the seed. random unit tests introduce something called flaky tests, which means that they may fail once in a while, and it's very hard to backtrack in which particular situation it fails

in addition, using default random number generator is in a sense OK for unit tests, but it's a biased number generator, which means that there is a higher probability to get some (series of) values than it is to get others. https://en.wikipedia.org/wiki/Random_number_generation#Uniform_distributions provides a good read-up on that.
 
this is seen as a security risk, especially when used in production code. It is less so for unit tests, but it's considered good practice to treat your test code as production code for various reasons. hence, great caution is necessary

therefore, the other test case is better

```suggestion
```